### PR TITLE
SDIT-1816 Return latest booking flag

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisonerprofile/PrisonerProfileService.kt [hunk 1]
```diff
@@ -22,6 +22,7 @@ class PrisonerProfileService(
           bookingId = it.bookingId,
           startDate = it.bookingBeginDate.toLocalDate(),
           endDate = it.bookingEndDate?.toLocalDate(),
+          latestBooking = it.bookingSequence == 1,
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisonerprofile/PrisonerProfileService.kt:25]: This is the main change - return a boolean indicating if this is the latest booking.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/Offender.kt [hunk 2]
```diff
@@ -107,13 +107,14 @@ class OffenderBuilder(
...
     livingUnitId: Long,
+    bookingSequence: Int?,
     dsl: BookingDsl.() -> Unit,
   ) = bookingBuilderFactory.builder()
     .let { builder ->
       builder.build(
         offender = offender,
         // if you want multiple bookings then create them with latest booking first so it gets seq 1 like in Nomis
-        bookingSequence = offender.bookings.size + 1,
+        bookingSequence = bookingSequence ?: offender.bookings.size + 1,
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/Offender.kt:117]: A quick hack to allow us to specify the booking sequence - works around an issue with the DSL where the booking sequence on both the main booking and an alias booking are 1. This should go away when a proper fix is done (currently WIP).

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/Offender.kt [hunk 3]
```diff
@@ -107,13 +107,14 @@ class OffenderBuilder(
...
     livingUnitId: Long,
+    bookingSequence: Int?,
     dsl: BookingDsl.() -> Unit,
   ) = bookingBuilderFactory.builder()
     .let { builder ->
       builder.build(
         offender = offender,
         // if you want multiple bookings then create them with latest booking first so it gets seq 1 like in Nomis
-        bookingSequence = offender.bookings.size + 1,
+        bookingSequence = bookingSequence ?: (offender.bookings.size + 1),
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/Offender.kt:117]: A quick hack to allow us to specify the booking sequence - works around an issue with the DSL where the booking sequence on both the main booking and an alias booking are 1. This should go away when a proper fix is done (currently WIP).

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**